packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as path from "path";
+import { readdirSync } from "fs";
+import { strict as assert } from "assert";
+import {
+	AcceptanceCondition,
+	createWeightedAsyncGenerator as createWeightedGenerator,
+	AsyncGenerator as Generator,
+	takeAsync as take,
+} from "@fluid-internal/stochastic-test-utils";
+import {
+	createDDSFuzzSuite,
+	DDSFuzzModel,
+	DDSFuzzHarnessEvents,
+} from "@fluid-internal/test-dds-utils";
+import { PropertySet } from "@fluidframework/merge-tree";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { IntervalCollection, SequenceInterval } from "../intervalCollection";
+import { SharedStringFactory } from "../sequenceFactory";
+import { SharedString } from "../sharedString";
+import {
+	appendAddIntervalToRevertibles,
+	// appendChangeIntervalToRevertibles,
+	// appendDeleteIntervalToRevertibles,
+	appendIntervalPropertyChangedToRevertibles,
+	appendSharedStringDeltaToRevertibles,
+	SharedStringRevertible,
+} from "../revertibles";
+import { assertEquivalentSharedStrings } from "./intervalUtils";
+import {
+	Operation,
+	RangeSpec,
+	AddText,
+	RemoveRange,
+	AddInterval,
+	DeleteInterval,
+	ChangeInterval,
+	ChangeProperties,
+	FuzzTestState,
+	makeRevertibleReducer,
+	RevertSharedStringRevertibles,
+	RevertOperation,
+} from "./intervalCollection.fuzzUtils";
+import { minimizeTestFromFailureFile } from "./intervalCollection.fuzzMinimization";
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+	/**
+	 * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+	 * Note due to concurency, during test execution the actual length of the string may exceed this.
+	 */
+	maxStringLength?: number;
+	/**
+	 * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+	 * Note due to concurency, during test execution the actual number of intervals may exceed this.
+	 */
+	maxIntervals?: number;
+	maxInsertLength?: number;
+	intervalCollectionNamePool?: string[];
+	propertyNamePool?: string[];
+	validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+	maxStringLength: 1000,
+	maxIntervals: 100,
+	maxInsertLength: 10,
+	intervalCollectionNamePool: ["comments"],
+	propertyNamePool: ["prop1", "prop2", "prop3"],
+	validateInterval: 100,
+};
+
+const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+const revertibles: SharedStringRevertible[] = [];
```

Comment:
the scope of this object looks off. I would have expected that you'd need to create a separate revertibles list for each client--presumably client 1 shouldn't be undoing client 2's work. It also looks like there's a separate declaration for revertibles in `intervalCollection.fuzzUtils.ts` used for performing the revert, which seems like it should be shared.

I think the "right" place for this is within the fuzz test's global state object, but there's not a super clean way to make the revertibles end up directly on that object. A simple approach might be to use this hook to set up an extra field on the client object (or maybe the SharedString) for that client's revertibles stack. Then you could reference that later from within your reducer. You'll need to do some casting though (i'd probably validate at runtime since the error should be clear)

Reply:
This might be what's contributing to the failures since trying to revert one client's changes on another client won't work for the revertibles that make local refs and tracking groups. (and is not supported)